DESIGN.md

```diff
@@ -66,9 +66,18 @@ Consumer of events without flow control.
 
 ##### Subject
 
-A "hot" data source that allows a producer to emit events and consumers to receive events in a multicast manner.
+A "hot", push-based data source that allows a producer to emit events to it and consumers to subscribe to events in a multicast manner. It is "hot" because consumers subscribing to it does not cause side-effects, or affect the data flow in any way. It is push-based and reactive because the producer is fully in charge. 
 
-It is "hot" because consumers subscribing to it does not cause side-effects, or affect the data flow in any way. It is push and reactive because the producer is fully in charge.
+Relation to Reactive Streams
+
+- It can not implement Reactive Streams `Publisher` unless it is created with a default flow control strategy. 
+- It can not implement `Processor` since a `Processor` must compose `request(n)` which can not be done with multicasting or pure push. 
```

Comment:
> 4.1: A Processor represents a processing stage—which is both a Subscriber and a Publisher and MUST obey the contracts of both.

Thus, it has to obey the rules of a `Publisher` which can't overflow its subscribers.

I strongly argued against `Processor` as it looks like a `Subject` but can't be used as one. It is actually more like the RxJava `Operator`, but requires subscription rather than lifting into the observer chain.